truss/templates/server/model_wrapper.py

```diff
@@ -177,16 +178,20 @@ async def preprocess(
             return payload
 
         if inspect.iscoroutinefunction(self._model.preprocess):
-            return await self._model.preprocess(payload)
+            return await _intercept_exceptions_async(self._model.preprocess)(payload)
         else:
-            return await to_thread.run_sync(self._model.preprocess, payload)
+            return await to_thread.run_sync(
+                _intercept_exceptions_sync(self._model.preprocess), payload
+            )
 
     def _predict_sync_with_error_handling(self, payload):
         try:
             return self._model.predict(payload)
         except Exception:
             logging.exception("Exception while running predict")
-            return {"error": {"traceback": traceback.format_exc()}}
+            raise HTTPException(
+                status_code=500, detail={"traceback": traceback.format_exc()}
```

Comment:
I am not sure that we should be return stack traces in the response. I think logging them is enough for debugging, then the exception should be a plain old 500

I think it's kind of a security risk to leak code in responses.